src/schema/sources.ts

```diff
@@ -1278,6 +1313,30 @@ export const resolvers: IResolvers<any, Context> = {
 
       return getSourceById(ctx, info, sourceId);
     },
+    hideSourceFeedPosts: async (_, { sourceId }: { sourceId: string }, ctx) => {
+      await ensureSourcePermissions(ctx, sourceId, SourcePermissions.View);
```

Comment:
Optional, but you could abstract this into a top level function as only the value is changing, example is [this](https://github.com/dailydotdev/daily-api/pull/1336/files#diff-e3ad1f5d3fe340ded80e09a15a95f7d80be472d110fe2bdbbce21dba440b925aR883).